### PR TITLE
Require explicit inclusion of custom op support in a minimal build

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Remove support from custom ops from the base minimal build as they contribute too much binary growth to an Android build.

Add ability to explicitly enable custom op support in a minimal build.
Change one minimal build CI to test adding custom op support (unit tests are run in that build to validate)

**Motivation and Context**
Reduce base minimal build size on Android.

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -445,7 +445,10 @@ def parse_arguments():
                         "See /docs/ONNX_Runtime_Format_Model_Usage.md for more information. "
                         "RTTI is automatically disabled in a minimal build. "
                         "To enable execution providers that compile kernels at runtime (e.g. NNAPI) pass 'extended' "
-                        "as a parameter. e.g. '--minimal_build extended'.")
+                        "as a parameter. e.g. '--minimal_build extended'. "
+                        "To enable support for custom operators pass 'custom_ops' as a parameter. "
+                        "e.g. '--minimal_build custom_ops'. This can be combined with an 'extended' build by passing "
+                        "'--minimal_build extended,custom_ops'")
```
Comment (edgchen1) [inline tools/ci_build/build.py:451]: >extended,custom_ops [](start = 42, length = 19)

how about something like `add_argument("--minimal_build", nargs="*", default=None, type=str.lower, help="...")`?
then args.minimal_build is [optional parameters] (meaning minimal build with optional config), or None (meaning no minimal build). and we can check for exact string match of the parameters

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 2]
```diff
@@ -439,11 +439,13 @@ static ORT_STATUS_PTR CreateSessionAndLoadModel(_In_ const OrtSessionOptions* op
         env->GetEnvironment());
   }
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
   // Add custom domains
   Status status;
```
Comment (edgchen1) [inline onnxruntime/core/session/onnxruntime_c_api.cc:444]: >Status status; [](start = 2, length = 14)

looks like this is not used

**Review by edgchen1 — DISMISSED** [review]
:shipit:

**Review by gwang-msft — APPROVED** [review]
:shipit:

**Review by gwang-msft — APPROVED** [review]

File: tools/ci_build/build.py [hunk 3]
```diff
@@ -1374,7 +1379,7 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
 
             # Disable python tests in a reduced build as we don't know which ops have been included and which
             # models can run
-            if args.include_ops_by_config or args.minimal_build != 'off':
+            if args.include_ops_by_config or args.minimal_build:
```
Comment (edgchen1) [inline tools/ci_build/build.py:1382]: one more place to update to `args.minimal_build is not None`

**Pull request merged**